src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+        assertEquals(actual, registeredUser);
+    }
+
+    @Test
+    void register_userWithShortLogin_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reex");
+        actual.setPassword("qwerty");
+        actual.setAge(25);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithNullLogin_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setPassword("qwerty");
+        actual.setAge(25);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithShortPassword_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reex");
+        actual.setPassword("qwer");
+        actual.setAge(25);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithNullPassword_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reex");
+        actual.setPassword(null);
+        actual.setAge(25);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void regixter_userWithCorrectLengthPassword_ok() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reexXx");
+        actual.setPassword("123456789");
+        actual.setAge(25);
+        String expected = "123456789";
+        assertEquals(actual.getPassword(), expected);
+
+    }
+
+    @Test
+    void register_userWithInvalidAge_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reex");
+        actual.setPassword("qwerty");
+        actual.setAge(17);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithNullAge_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reex");
+        actual.setPassword("qwerty");
+        actual.setAge(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithNegativeAge_notOk() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reexXx");
+        actual.setPassword("qwerty");
+        actual.setAge(-30);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_userWithValidAge() {
+        User actual = new User();
+        actual.setId(111L);
+        actual.setLogin("reexXx");
+        actual.setPassword("qwerty");
+        actual.setAge(25);
+        int expected = 25;
+        assertEquals(actual.getAge(), expected);

```

Comment:
So you tested here that creating object in Java works well? )))) 

You need to register user and then check is age the same 

User actualUser = registrationService.register(user));
assertNotNull(actualUser);
assertNotNull(actualUser.getId());
assertEquals(user, actualUser);

Reply:
But I can't register actualUser with user logins.
To be honest, I don't understand why we need

assertNotNull(actualUser);
assertNotNull(actualUser.getId());
assertEquals(user, actualUser);

here

Reply:
Discussed offline